src/super_gradients/common/sg_loggers/clearml_sg_logger.py

```diff
@@ -0,0 +1,226 @@
+    def add_scalar(self, tag: str, scalar_value: float, global_step: int = 0):
+        super(ClearMLSGLogger, self).add_scalar(tag=tag, scalar_value=scalar_value, global_step=global_step)
+        self.__add_scalar(tag=tag, scalar_value=scalar_value, global_step=global_step)
+
+    @multi_process_safe
+    def add_scalars(self, tag_scalar_dict: dict, global_step: int = 0):
+        super(ClearMLSGLogger, self).add_scalars(tag_scalar_dict=tag_scalar_dict, global_step=global_step)
+        for tag, scalar_value in tag_scalar_dict.items():
+            self.__add_scalar(tag=tag, scalar_value=scalar_value, global_step=global_step)
+
+    def __add_image(
+        self,
+        tag: str,
+        image: Union[torch.Tensor, np.array, Image.Image],
+        global_step: int,
+    ):
+        if isinstance(image, torch.Tensor):
+            image = image.cpu().detach().numpy()
+        if image.shape[0] < 5:
+            image = image.transpose([1, 2, 0])
+        self.clearml_logger.report_image(
+            title=tag,
+            series=tag,
+            image=image,
+            iteration=global_step,
+            max_image_history=-1,
+        )
+
+    @multi_process_safe
+    def add_image(
+        self,
+        tag: str,
+        image: Union[torch.Tensor, np.array, Image.Image],
+        data_format="CHW",
+        global_step: int = 0,
+    ):
+        super(ClearMLSGLogger, self).add_image(tag=tag, image=image, data_format=data_format, global_step=global_step)
+        self.__add_image(tag, image, global_step)
+
+    @multi_process_safe
+    def add_images(
+        self,
+        tag: str,
+        images: Union[torch.Tensor, np.array],
+        data_format="NCHW",
+        global_step: int = 0,
+    ):
+        super(ClearMLSGLogger, self).add_images(tag=tag, images=images, data_format=data_format, global_step=global_step)
+        for image in images:
+            self.__add_image(tag, image, global_step)
+
+    @multi_process_safe
+    def add_video(self, tag: str, video: Union[torch.Tensor, np.array], global_step: int = 0):
+        super().add_video(tag, video, global_step)
+        logger.warning("ClearMLSGLogger does not support uploading video to clearML from a tensor/array.")
+
+    @multi_process_safe
+    def add_histogram(
+        self,
+        tag: str,
+        values: Union[torch.Tensor, np.array],
+        bins: str,
+        global_step: int = 0,
+    ):
+        super().add_histogram(tag, values, bins, global_step)
+        self.clearml_logger.report_histogram(title=tag, series=tag, iteration=global_step, values=values)
+
+    @multi_process_safe
+    def add_text(self, tag: str, text_string: str, global_step: int = 0):
+        super().add_text(tag, text_string, global_step)
+        self.clearml_logger.report_text(text_string)
+
+    @multi_process_safe
+    def add_figure(self, tag: str, figure: plt.figure, global_step: int = 0):
+        super().add_figure(tag, figure, global_step)
+        logger.warning(
```

Comment:
instead, just save the figure into a temp image and upload it as an image

Reply:
Added here: c99c2639fa3718156470d6b77a31091e8ac3ce37